k4EDM4hep2LcioConv/src/k4Lcio2EDM4hepConv.cpp

```diff
@@ -550,6 +537,44 @@ namespace LCIO2EDM4hepConv {
     return retColls;
   }
 
+  std::unique_ptr<edm4hep::CaloHitContributionCollection> createCaloHitContributions(
+    TypeMapT<const lcio::SimCalorimeterHit*, edm4hep::MutableSimCalorimeterHit>& SimCaloHitMap,
+    const TypeMapT<const lcio::MCParticle*, edm4hep::MutableMCParticle>& mcparticlesMap)
+  {
+    {
+      auto contrCollection = std::make_unique<edm4hep::CaloHitContributionCollection>();
+      for (auto& [lcioHit, edmHit] : SimCaloHitMap) {
+
+        auto NMCParticle = lcioHit->getNMCParticles();
+        for (unsigned j = 0; j < NMCParticle; j++) {
+          auto edm_contr = contrCollection->create();
+
+          edm_contr.setPDG(lcioHit->getPDGCont(j));
+          edm_contr.setTime(lcioHit->getTimeCont(j));
+          edm_contr.setEnergy(lcioHit->getEnergyCont(j));
+          edm_contr.setStepPosition(lcioHit->getStepPosition(j));
+          auto lcioParticle = (lcioHit->getParticleCont(j));
+          if (lcioParticle == nullptr) {
+            edm_contr.setParticle(nullptr);
+          }
+          else {
+            const auto it = mcparticlesMap.find(lcioParticle);
+            if (it != mcparticlesMap.end()) {
+              edm_contr.setParticle(it->second);
+            }
+            else {
+              std::cerr << "Cannot find corresponding EDM4hep MCParticle for the LCIO MCParticle, "
+                           "while trying to resolve the parents of MCParticles Collections"
```

Comment:
```suggestion
                           "while trying to build CaloHitContributions"
```
probably copy-past mistake.